### PR TITLE
Enforce import order with ESLint

### PR DESCRIPTION
**Pull request by renchap**
And I am here again with a big automated refactor!

In this PR, I enabled automatic re-ordering of `import` statements, to be able to group them together by similar groups, and then order them alphabetically inside each group.

It makes them consistent over files, make it easy to see what a file uses and easier to find the `import` line corresponding to the function you are looking for.

This was already applied for Typescript files, but this PR enabled it for every JS file.

First commit changes the ESLint config, second one is `yarn test:js:lint --fix` ran multiple times until it does not do any automated fix, then a few manual fixes on files that `eslint` did not manage to correct.

To understand how I specified the grouping, here is a recent count of all external imports (imported more than 1 time):

```
   2 fuzzysort
   2 history
   2 react-router-scroll-4
   2 react-sparklines
   2 stacktrace-js
   3 intl-messageformat
   3 react-swipeable-views
   3 react-textarea-autosize
   3 stringz
   4 axios
   4 escape-html
   4 react-redux-loading-bar
   4 react-toggle
   5 react-hotkeys
   5 react-test-renderer
   7 react-dom
   8 detect-passive-events
  11 reselect
  29 lodash
  32 react-helmet
  32 react-router-dom
  62 classnames
  72 immutable
  75 react-immutable-pure-component
 108 react-immutable-proptypes
 140 react-redux
 179 react-intl
 194 prop-types
 243 react
```

### CODE REVIEW

File: app/javascript/mastodon/actions/notifications.js [hunk 1]
```diff
@@ -1,5 +1,16 @@
-import api, { getLinks } from '../api';
 import IntlMessageFormat from 'intl-messageformat';
+import { defineMessages } from 'react-intl';
+
+import { List as ImmutableList } from 'immutable';
+
+
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/actions/notifications.js:6]: There are lots of blank lines, that does look weird and unintended.

File: app/javascript/mastodon/actions/notifications.js [hunk 2]
```diff
@@ -9,12 +20,11 @@ import {
...
-import { List as ImmutableList } from 'immutable';
-import { unescapeHTML } from '../utils/html';
-import { usePendingItems as preferPendingItems } from 'mastodon/initial_state';
-import { compareId } from 'mastodon/compare_id';
-import { requestNotificationPermission } from '../utils/notifications';
+
+
+
+
+
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/actions/notifications.js:27]: More blank lines!

File: app/javascript/mastodon/components/admin/Counter.jsx [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 import { FormattedNumber } from 'react-intl';
-import { Sparklines, SparklinesCurve } from 'react-sparklines';
+
 import classNames from 'classnames';
-import { Skeleton } from 'mastodon/components/skeleton';
+
+import { Sparklines, SparklinesCurve } from 'react-sparklines';
+
+import api from 'mastodon/api';
+import {Skeleton} from 'mastodon/components/skeleton';
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/components/admin/Counter.jsx:12]: The lack of spaces around `Skeleton` is surprising and inconsistent with the remaining of the code. What's causing this?

**Review by ClearlyClaire — COMMENTED** [review]
This introduces a lot of blank lines and a few other stylistic inconsistencies. I wonder what may be causing this?

File: app/javascript/mastodon/components/admin/Counter.jsx (hunk 3 above)
Comment (renchap) [reply app/javascript/mastodon/components/admin/Counter.jsx:12]: My manual rebase before posting the PR, and the fact that this is not auto-fixed by Prettier.

Will go through them and fix this (same for newlines)

**Comment by renchap** [comment]
Blank lines were somehow kept by `eslint` when re-ordering, sometimes between groups, sometimes after the imports. Usually those are removed by Prettier, but as Prettier is not applied on the JS files, it did not happen. We can also have the [`no-multiple-empty-lines`](no-multiple-empty-lines) rule enabled, but thats not the case.

Pushed a new commit which removes them using a regexp.

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**